### ISSUE TITLE
Random normal deviates

### ISSUE THREAD

**Issue by Beliavsky** [j3-fortran/fortran_proposals#210]
Random normal deviates are commonly needed and are available in many other scientific computing languages, such as Matlab, R, Python/Numpy, Julia, and C++, so I think Fortran should also have a random_normal() subroutine, analogous to random_number().

**Comment by certik** [comment]
I think this would be a good fit for [stdlib](https://github.com/fortran-lang/stdlib/), would you mind proposing it there?

**Comment by Beliavsky** [comment]
I think people are already working on non-uniform RNG, including normal deviates, in stdlib, for example https://github.com/fortran-lang/stdlib/pull/240 . There are many known statistical distributions to sample from, and I'm not sure that everything that will be implemented in stdlib should be in the standard, but I think normal variates should be.

**Comment by Beliavsky** [comment]
According to a [post by John Cook](https://www.johndcook.com/blog/cpp_tr1_random/), the C++ standard includes

Generating from non-uniform distributions
Distributions directly supported
Bernoulli
Binomial
Exponential
Gamma
Geometric
Normal (Gaussian)
Poisson

but not
Cauchy
Chi squared
Student t
Snedecor F
Weibull
Rayleigh

Distributions that can be sampled easily, if not very efficiently, given a normal RNG, may not be needed in the standard -- examples are the Student's t and chi-squared distributions.

**Comment by Beliavsky** [comment]
Non-uniform RNG in [R](https://stat.ethz.ch/R-manual/R-patched/library/stats/html/Distributions.html), [NumPy](https://numpy.org/doc/stable/reference/random/generator.html#numpy.random.Generator) [SciPy](https://docs.scipy.org/doc/scipy/reference/stats.html), [Matlab statistics toolbox](https://www.mathworks.com/help/stats/supported-distributions.html)

**Comment by peteroupc** [comment]
> According to a [post by John D. Cook](https://www.johndcook.com/blog/cpp_tr1_random/), the C++ standard includes
> 
> Generating from non-uniform distributions
> Distributions directly supported
> Bernoulli
> Binomial
> Exponential
> Gamma
> Geometric
> Normal (Gaussian)
> Poisson
> 
> but not
> Cauchy
> Chi squared
> Student t
> Snedecor F
> Weibull
> Rayleigh

That post refers to the Technical Report 1 extensions of C++, which predated C++11 which introduced `<random>` and its distributions for the first time in the main standard.  In C++11, of the distributions listed here, only the Rayleigh distribution is not offered there.

**Comment by johnwcowan** [comment]
The [proposal for the next Scheme standard](https://srfi.schemers.org/srfi-194/srfi-194.html)
specifies the following generators:

#### Uniform generators

* arbitrary integer bounds
* integer bounds in the ranges of `uint1 uint8 int8 uint16 int16 uint32 int32 uint64 int64
* real numbers with arbitrary bounds
*  complex numbers with arbitrary rectangular or polar bounds
*  booleans
*  characters drawn from a string with replacement
* strings whose characters are drawn from a string with replacement and whose lengths are uniformly distributed between 0 and an upper bound.

#### Non-uniform distributions
* Bernoulli
*  binomial
*  categorical (given a vector of weights)
* normal
*  exponential
* geometric
*  Poisson
*  Zipf.  

#### Generators of fixed-length vectors
* uniformly from the surface or interior of an *n*-sphere
* uniformly from the surface of an *n*-ellipsoid
